src/linker/Linker/Driver.cs

```diff
@@ -161,437 +163,453 @@ static void ErrorMissingArgument (string optionName)
 			Console.WriteLine ($"Missing argument for '{optionName}' option");
 		}
 
-		public bool Run (ILogger customLogger = null)
```

Comment:
Could you keep the code as it and instead in the mock type override GetDefaultContext to extract the context instance?

Reply:
I don't think that works. I want to get the context after command-line arguments have been processed, but without trying to resolve assemblies/xml files from disk. So I need to separate the argument processing into something I can run separately from processing the steps.

Originally I think I misunderstood your suggestion - I thought you meant that GetDefaultContext could be used to inject mocked things into the context. So I tried to set up a pipeline with mocked steps that would do nothing when processed. That just created more problems though, requiring more extension points in LinkContext, and more machinery to get information out of the steps after `Run` finishes - especially because `Pipeline.Process` removes steps when it runs.

Reply:
The context is a reference instance you can store it and when referenced later it will give you the right data